src/usermod.c

```diff
@@ -2153,7 +2153,7 @@ int main (int argc, char **argv)
 #endif
 
 	sys_ngroups = sysconf (_SC_NGROUPS_MAX);
-	user_groups = (char **) malloc (sizeof (char *) * (1 + sys_ngroups));
+	user_groups = (char **) xmalloc (sizeof (char *) * (1 + sys_ngroups));
 	user_groups[0] = (char *) 0;
```

Comment:
Can we please write this as `NULL`? :-)

Reply:
Can we also write it without that ugly cast?

Reply:
I opened a PR for that: <https://github.com/shadow-maint/shadow/pull/642>